todo_app/templates/index.html

```diff
@@ -24,8 +24,9 @@ <h1 class="display-4">To-Do App</h1>
                   <option value="Done">Done</option>
                 </select>
               </div>
-
-              <button type="submit" class="btn btn-primary"><b>Create Task</b></button>
+              {% if view_model.user_role == 'writer' %}
+                <button type="submit" class="btn btn-primary"><b>Create Task</b></button>
```

Comment:
I'd also be tempted to wrap the whole form in this if statement: as it is it gives users a _very_ strong hint what the endpoint is for this, and what data it expects, so even though browser-side protections are very easily bypassed, this does make it much easier for them to check whether you've got server-side protections